js/components/util/files.ts

```diff
@@ -122,55 +110,39 @@ export const loadHttpFile = async (
+        const response = await fetch(url);
+        if (response.ok) {
+            const reader = response.body!.getReader();
+            let received = 0;
+            const chunks: Uint8Array[] = [];
 
-                    return reader.read().then(
-                        function readChunk(result): Promise<ArrayBufferLike> {
-                            if (result.done) {
-                                const data = new Uint8Array(received);
-                                let offset = 0;
-                                for (let idx = 0; idx < chunks.length; idx++) {
-                                    data.set(chunks[idx], offset);
-                                    offset += chunks[idx].length;
-                                }
-                                return Promise.resolve(data.buffer);
-                            }
+            let result = await reader.read();
+            while (!result.done) {
+                chunks.push(result.value);
+                received += result.value.length;
+                result = await reader.read();
+            }
 
-                            received += result.value.length;
-                            // if (contentLength) {
-                            //     loadingProgress(received, contentLength);
-                            // }
-                            chunks.push(result.value);
+            const data = new Uint8Array(received);
+            let offset = 0;
+            for (const chunk of chunks) {
+                data.set(chunk, offset);
+                offset += chunk.length;
+            }
 
-                            return reader.read().then(readChunk);
-                        });
-                } else {
-                    reject('Error loading: ' + response.statusText);
-                }
-            }).then(function (data: ArrayBufferLike) {
-                const urlParts = url!.split('/');
-                const file = urlParts.pop()!;
-                const fileParts = file.split('.');
-                const ext = fileParts.pop()!.toLowerCase();
-                const name = decodeURIComponent(fileParts.join('.'));
-                if (data.byteLength >= 800 * 1024) {
-                    reject(`Unable to load ${url}`);
-                } else if (
-                    includes(NIBBLE_FORMATS, ext) &&
-                    disk2.setBinary(number, name, ext, data)
-                ) {
-                    initGamepad();
-                } else {
-                    reject(`Extension ${ext} not recognized.`);
-                }
-                resolve(true);
-            }).catch(reject);
-        });
+            const urlParts = url!.split('/');
+            const file = urlParts.pop()!;
+            const fileParts = file.split('.');
+            const ext = fileParts.pop()!.toLowerCase();
+            const name = decodeURIComponent(fileParts.join('.'));
+            if (includes(NIBBLE_FORMATS, ext)) {
```

Comment:
I don't mind so much here, but if you change the other ones, consider changing this one, too.